utils.go

```diff
@@ -81,6 +82,21 @@ func newStorager(pairs ...types.Pair) (store *Storage, err error) {
 		store.workDir = opt.WorkDir
 	}
 
+	if opt.HasCredential {
```

Comment:
Please don't add not related changes in this PR.

- If credential not supported, please add support in new PRs than merge master.
- If already supported, please merge master and update the branch.